webhook_server_container/libs/github_api.py

```diff
@@ -628,12 +614,13 @@ def assign_reviewers(self) -> None:
         self.logger.info(f"{self.log_prefix} Assign reviewers")
         changed_files = self.list_changed_commit_files()
         reviewers_to_add = self.reviewers
-        for _file, _file_reviewers in self.files_reviewers.items():
-            if _file in changed_files:
-                reviewers_to_add.extend(_file_reviewers)
-        for _folder, _folder_reviewers in self.folders_reviewers.items():
-            if any(cf for cf in changed_files if _folder in str(Path(cf).parent)):
-                reviewers_to_add.extend(_folder_reviewers)
+        changed_folders = [Path(cf).parent for cf in changed_files]
+
+        for changed_folder_path in changed_folders:
+            for owners_path, owners_data in self.approvers_and_reviewers.items():
```

Comment:
Say the PR changes file in  folder 1 and folder 2, both with different OWNERS and under root folder. In this case only approvers should be their parent(root's) OWNERS file.

Reply:
I don't understand what you mean.